### PR TITLE
Bug 1392761 - Pull Down in Bookmarks to Trigger a Bookmark Sync Refresh

### PR DESCRIPTION
**Pull request by joeyg**
This PR adds pull to refresh to the Bookmarks screen. If the account is synced this triggers a sync and reloads the data from the local database.

Here is the screenshot gif: http://g.recordit.co/lg4EsHknpL.gif

### CODE REVIEW

File: Client/Frontend/Home/BookmarksPanel.swift [hunk 1]
```diff
@@ -75,11 +76,25 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
         tableView.addGestureRecognizer(longPressRecognizer)
 
         self.tableView.accessibilityIdentifier = "Bookmarks List"
+        
+        self.refreshControl = UIRefreshControl()
+        self.tableView.addSubview(refreshControl!)
```
Comment (jhugman) [inline Client/Frontend/Home/BookmarksPanel.swift]: nit: remove `self`.

File: Client/Frontend/Home/BookmarksPanel.swift [hunk 2]
```diff
@@ -110,6 +125,15 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
             break
         }
     }
+    
+    @objc fileprivate func refreshBookmarks() {
+        profile.syncManager.mirrorBookmarks().upon { (_) in
+            DispatchQueue.main.async {
+                self.loadData()
```
Comment (rnewman) [inline Client/Frontend/Home/BookmarksPanel.swift]: What happens if you pull to refresh when you're in a subfolder?

What happens if that subfolder is deleted as a result of the sync?

@joeyg, @jhugman — did you test this?

Comment (joeyg) [reply Client/Frontend/Home/BookmarksPanel.swift]: @rnewman I tested deleting the folder that is selected and all that is displayed is the Header linking to the parent folder. I have not yet tested deleting the parent folder of the current folder - I can take a look that when I have some time.

**Comment by joeyg** [comment]
@rnewman Here is a screencast of the functionality for removing folders. In it I navigate to the inner folder, delete the outer folder, pull to refresh and then navigate back out. http://g.recordit.co/DpOHtfwrGQ.gif

It seems like a reasonable experience to me but let me know if the user experience should be something else.

**Comment by rnewman** [comment]
Good enough for me, @joeyg — so long as the user isn't stranded, I'm happy.

**Comment by joeyg** [comment]
@jhugman Let me know if you need some additional changes to this PR.

**Comment by joeyg** [comment]
@rnewman I added a Kif test for this feature but let me know if it XCUITest is preferred.

**Comment by rnewman** [comment]
That sounds like a question for @jhugman!

**Comment by joeyg** [comment]
Oops! Sorry for the ping

**Comment by jhugman** [comment]
@joeyg – sorry this has taken a little while to get back to. 

> but let me know if it XCUITest is preferred.

We're trying to move all our UI tests to XCUITests now. However, the state of the app due to Photon changes means that the tests have not kept up. 

@npark-mozilla is the person who will be leading that charge. I hope it's going to be a relatively simple (if not large) job.

Let's land this now, and get it shipped in time for 10.0. We can reup the tests when the photon work settles down.

**Comment by joeyg** [comment]
@jhugman - thanks for the clarification on the tests. I’ll make sure to make them XCUITests in the future.

**Comment by jhugman** [comment]
@joeyg – sorry this has taken a little while to get back to. 

> but let me know if it XCUITest is preferred.

We're trying to move all our UI tests to XCUITests now. However, the state of the app due to Photon changes means that the tests have not kept up. 

@npark-mozilla is the person who will be leading that charge. I hope it's going to be a relatively simple (if not large) job.

Let's land this now, and get it shipped in time for 10.0. We can reup the tests when the photon work settles down.

**Comment by jhugman** [comment]
I'd missed that this branch wasn't getting built by BuddyBuild (a prerequisite to landing this PR)

**Pull request closed without merging**